src/Octopurls.Tests/RedirectsURLsTests.cs

```diff
@@ -0,0 +1,140 @@
+                select url.Value;
+
+            badURLs.AddRange(query.ToList());
+
+            if (badURLs.Any())
+            {
+                Console.WriteLine("The bad urls are:");
+                foreach (var badUrL in badURLs)
+                {
+                    Console.WriteLine($"{badUrL}");
+                }
+            }
+
+            Assert.Equal(0, badURLs.Count());
+        }
+
+        [Fact]
+        public void TestURLMethodFailsWithBadURLs()
+        {
+            var url = "https://steam.com/purchase/HalfLife2Ep3";
+            Assert.False(TestURL(url), $"Web request for the fake URL {url} did not fail, but it should have");
+        }
+
+        private bool TestURL(string url)
+        {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            request.Timeout = 15000;
+            request.AllowAutoRedirect = true;
+            request.UseDefaultCredentials = true;
+
+            request.Method = "HEAD";
+
+            try
+            {
+                //First trying with a HEAD request
+                using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
+                {
+                    return ValidateResponse(response);
+                }
+            }
+            catch (WebException e)
+            {
+                try
+                {
+                    HttpWebRequest secondrequest = (HttpWebRequest)WebRequest.Create(url);
+                    secondrequest.Timeout = 15000;
+                    secondrequest.AllowAutoRedirect = true;
+                    secondrequest.UseDefaultCredentials = true;
+
+                    secondrequest.Method = "GET";
+
+                    //If the request for HEAD fails, try GET which is a bit more expensive. Lots of Microsoft links for some reason blow up doing a HEAD, but succeed with GET
+                    using (HttpWebResponse response = (HttpWebResponse)secondrequest.GetResponse())
+                    {
+                        return ValidateResponse(response);
+                    }
+
+                }
+
+                //If both HEAD and GET fail, then its definitely a bad URL 
+                catch (WebException exception)
+                {
+                    //In this case the WebException doesn't return the status code, so we need to read it from the exception.Message
+                    foreach (var code in _acceptedStatusCodesOver400)
+                    {
+                        if (exception.Message.Contains(code))
+                        {
+                            Console.WriteLine($"Success - Url [{url}] returned status code [{code}] which is in the list of accepted codes over 400");
+                            return true;
+                        }
+                    }
+
+                    Console.WriteLine($"Failure - URL [{url}] returned error [{exception.Message}]");
+                    return false;
+                }
```

Comment:
this might be better using LINQ too, e.g.

```
var acceptableStatusCode = _acceptedStatusCodesOver400.FirstOrDefault(c => exception.Message.Contains(c));
if (acceptableStatusCode != null)
{
    Console.WriteLine($"Success - Url [{url}] returned status code [{acceptableStatusCode }] which is in the list of accepted codes over 400");
     return true;
}
```